tests/instruments/racal/test_racal1992.py

```diff
@@ -0,0 +1,339 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from pymeasure.test import expected_protocol
+from pymeasure.instruments.racal import Racal1992
+
+# ============================================================
+# TESTS
+# ============================================================
+
+
+def test_self_check():
+    with expected_protocol(
+            Racal1992,
+            [
+                (' CK', 'CK+010.00000000E+06'),
+            ],
+    ) as instr:
+        instr.operating_mode = 'self_check'
+        assert instr.measured_value == 10000000.0
```

Comment:
Changing the operating mode doesn't result in a guaranteed response. But once an operating mode is set, the instrument will continuously store measurement data in its response data FIFO. In the case of self_check, the instrument is configured with a 10MHz input that measures frequency. If the resolution is set to 9 digits or less, this will result in a new measurement at least every second, so reading a value will already result in a measured_value.
I don't think anything needs to change here.